src/lib/RpcApi.ts

```diff
@@ -62,6 +62,16 @@ export default class RpcApi {
         return client;
     }
 
+    public routerPush(routeName: string) {
```

Comment:
They need to be accessible from every component, thus the Vue prototype property `.$rpc` lends itself perfectly to thus oppertunity, as otherwise we would have to extend the native `vue-router`, which would also be available in each component as `.$router`. These `routerPush` and `routerReplace` functions however are wrappers for the `vue-router`'s native `push` and `replace` methods. Thus it's a bit hard to come up with better names on the router instance itself, but it would still be prototype-extension, which is generally considered bad practice.

Reply:
And it also actually has to do with the RpcApi itself, because it's used for preserving the rpc parameters of the request and responses. Thus it's not _that_ bad a place for the methods.